### PR TITLE
Feature/fix perpetual futures net position of market

### LINKED ISSUE #479: [derivatives] Handle PerpetualFuturesNetPositionOfMarket properly to reflect the both long/short positions
the current implementation doesn't handle `PerpetualFuturesNetPositionOfMarket`correctly to reflect the total position size on the perpetual futures.
The code adds position size when long is taken and "deducts" position size when short is taken.
This way cannot be considered as the correct way to manage `PerpetualFuturesNetPositionOfMarket`.
Need the separation between long/short position for each market.

### PR DESCRIPTION
**Pull request by taiki1frsh**
close #479 

Besides, implemented ClosePosition unit test

### CODE REVIEW

File: proto/derivatives/perpetual_futures.proto [hunk 1]
```diff
@@ -98,7 +98,10 @@ message PerpetualFuturesNetPositionOfMarket {
     (gogoproto.moretags) = "yaml:\"market\"",
     (gogoproto.nullable) = false
   ];
-  string position_size_in_micro = 2 [
+  PositionType position_type = 2 [
+    (gogoproto.moretags) = "yaml:\"position_type\""
+  ];
+  string position_size_in_denom_unit = 3 [
```
[the remark this answers is not in the document]
Comment (taiki1frsh) [reply proto/derivatives/perpetual_futures.proto:104]: No, 
should it be `sdk.Dec`?

[the remark this answers is not in the document]
Comment (taiki1frsh) [reply proto/derivatives/perpetual_futures.proto:104]: But, it's treated as sdk.Int inside

[the remark this answers is not in the document]
Comment (KimuraYu45z) [reply proto/derivatives/perpetual_futures.proto:104]: Interface that is exposed to API users should be treated as well as other places. PositionSize is treated in `Dec` in Msg for example so should follow it.

[the remark this answers is not in the document]
Comment (taiki1frsh) [reply proto/derivatives/perpetual_futures.proto:104]: no, this type is not exposed as API at this moment

[the remark this answers is not in the document]
Comment (KimuraYu45z) [reply proto/derivatives/perpetual_futures.proto:104]: If so, it is ok to use `Int`.

just `denom_exponent` is better I think instead of `denom_unit`

**Review by KimuraYu45z — APPROVED** [review]

File: proto/derivatives/perpetual_futures.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (taiki1frsh) [reply proto/derivatives/perpetual_futures.proto:104]: yeah, it makes sense.
I will modify and merge this.

**Pull request merged**